test/local/contracts/EscrowDistributor.test.ts

```diff
@@ -0,0 +1,123 @@
+const INITIAL_SUPPLY = ethers.utils.parseUnits("313373");
+const DURATION_WEEKS = BigNumber.from(52);
+const APPROVAL_AMOUNT = wei(1000000).toBN();
+
+// test accounts
+let owner: SignerWithAddress;
+let addr0: SignerWithAddress;
+let addr1: SignerWithAddress;
+let addr2: SignerWithAddress;
+let TREASURY_DAO: SignerWithAddress;
+
+// core contracts
+let kwenta: Contract;
+let rewardEscrow: Contract;
+let distributor: Contract;
+
+describe("EscrowDistributor", () => {
+    beforeEach("Deploy contracts", async () => {
+        [owner, addr0, addr1, addr2, TREASURY_DAO] = await ethers.getSigners();
+
+        let deployments = await deployKwenta(
+            NAME,
+            SYMBOL,
+            INITIAL_SUPPLY,
+            owner,
+            TREASURY_DAO
+        );
+
+        kwenta = deployments.kwenta;
+        rewardEscrow = deployments.rewardEscrow;
+    });
+
+    it("sets escrow address on constructor", async () => {
+        const EscrowDistributor = await ethers.getContractFactory(
+            "EscrowDistributor"
+        );
+        distributor = await EscrowDistributor.deploy(
+            kwenta.address,
+            rewardEscrow.address
+        );
+
+        await distributor.deployed();
+        expect(await distributor.rewardEscrowAddr()).to.equal(
+            rewardEscrow.address
+        );
+    });
+
+    it("fails to batch escrow when number of accounts doesn't equal number of amounts", async () => {
+        const EscrowDistributor = await ethers.getContractFactory(
+            "EscrowDistributor"
+        );
+        distributor = await EscrowDistributor.deploy(
+            kwenta.address,
+            rewardEscrow.address
+        );
+        await distributor.deployed();
+        await expect(
+            distributor.distirbuteEscrowed(
+                [addr0.address],
+                [wei(100).toBN(), wei(1000).toBN()],
+                DURATION_WEEKS
+            )
+        ).to.be.revertedWith(
+            "Number of accounts does not match number of values"
+        );
+    });
+
+    it("distributes escrowed amounts successfully", async () => {
+        const EscrowDistributor = await ethers.getContractFactory(
+            "EscrowDistributor"
+        );
+        distributor = await EscrowDistributor.deploy(
+            kwenta.address,
+            rewardEscrow.address
+        );
+        await distributor.deployed();
+
+        await kwenta
+            .connect(TREASURY_DAO)
+            .transfer(owner.address, wei(10000).toBN());
+
+        await kwenta
+            .connect(owner)
+            .approve(distributor.address, APPROVAL_AMOUNT);
+
+        expect(
+            await distributor
+                .connect(owner)
+                .distirbuteEscrowed(
+                    [addr0.address, addr1.address, addr2.address],
+                    [wei(1).toBN(), wei(50).toBN(), wei(200).toBN()],
+                    DURATION_WEEKS
+                )
+        ).to.emit(distributor, "BatchEscrowed");
```

Comment:
Would be good to add test which ensure event params are correct

Reply:
is there a different function for this, it seems `emit` only takes contract and eventName?

Reply:
Yes there is. Here is an example I have used in the past but there is also an easier way that is a little less verbose

https://github.com/Kwenta/margin-manager/blob/906267e512f4242274235800f3e03370b04d0e7d/test/integration/margin.behavior.ts#L1420

Reply:
You can chain `emit()` with `withArgs(arg1, arg2, arg3...)`

https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events

Reply:
perfect, thanks